packages/rpc-core/src/rpc-methods/__tests__/simulate-transaction-test.ts

```diff
@@ -0,0 +1,717 @@
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('replaces the invalid blockhash when `replaceRecentBlockhash` is true', async () => {
+        expect.assertions(1);
+        const secretKey = await getSecretKey();
+        const message = getMockTransactionMessage({
+            blockhash: base58.deserialize(new Uint8Array(Array(32).fill(0)))[0],
+            feePayerAddressBytes: MOCK_PUBLIC_KEY_BYTES,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+        });
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                {
+                    commitment: 'processed',
+                    encoding: 'base64',
+                    replaceRecentBlockhash: true,
+                }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: null,
+                err: null,
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('throws when called with a transaction having an unsupported version', async () => {
+        expect.assertions(1);
+        const [secretKey, { value: latestBlockhash }] = await Promise.all([
+            getSecretKey(),
+            rpc.getLatestBlockhash().send(),
+        ]);
+        const message = getMockTransactionMessage({
+            blockhash: latestBlockhash.blockhash,
+            feePayerAddressBytes: MOCK_PUBLIC_KEY_BYTES,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+            version: 0xfe, // Version 126
+        });
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                { commitment: 'processed', encoding: 'base64' }
+            )
+            .send();
+
+        await expect(resultPromise).rejects.toMatchObject({
+            code: -32602 satisfies (typeof SolanaJsonRpcErrorCode)['JSON_RPC_INVALID_PARAMS'],
+            message: expect.stringContaining('expected a valid transaction message version'),
```

Comment:
This might need to change now. See https://github.com/solana-labs/solana-web3.js/pull/1534/files#diff-a9e1b8ed269c088358fa54dabec9128cf162fd8c1a1f91d09fd4d8a0ea5edff0